LogEntryType registry (#4235)
pretix/pretix: doc/development/api/plugins.rst

through ``pip``. During development, you can just run ``python setup.py develop`` inside
your plugin source directory to make it discoverable.

> .. _`Signals`:

Author: :+1:

```suggestion
.. _`signals`:
```

Note that ``installed`` will *not* be called if the plugin is indirectly activated for an event
because the event is created with settings copied from another event.

> .. _`Registries`:

Author: ```suggestion
.. _`registries`:
```

Registries
----------

Many signals in pretix are used so that plugins can "register" a class, e.g. a payment provider or a
ticket renderer.

However, for some of them (types of :ref:`Log Entries <logging>`) we use a different method to keep track of them:
In a ``Registry``, classes are collected at application startup, along with a unique key (in case
of LogEntryType, the action_type) as well as which plugin registered them.

To register a class, you can use one of several decorator provided by the Registry object:

.. code-block:: python

   @log_entry_types.new('my_pretix_plugin.some.action', _('Some action in My Pretix Plugin occured.'))
   class MyPretixPluginLogEntryType(EventLogEntryType):

>        pass

Author: I agree... I changed this to just include the code documentation for the decorators, instead of providing example code directly, because I couldn't think of a simpler example

> To register a class, you can use one of several decorator provided by the Registry object:
> 
> .. code-block:: python
> 
>    @log_entry_types.new('my_pretix_plugin.some.action', _('Some action in My Pretix Plugin occured.'))
>    class MyPretixPluginLogEntryType(EventLogEntryType):
>        pass

Author: ```suggestion
To register a class, you can use one of several decorators provided by the Registry object:

.. autoclass:: pretix.base.logentrytypes.LogEntryTypeRegistry
   :members: register, new, new_from_dict
```
